core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -209,24 +214,23 @@ public void injectDependencies(EvictionManager evictionManager,
    }
 
    private void assertKeyNotNull(Object key) {
-      if (key == null) {
-         throw new NullPointerException("Null keys are not supported!");
-      }
+      requireNonNull(key,"Null keys are not supported!");
```

Comment:
with my intellij version sometimes is hard to see if I don't format the code explicitly ! done